test/expect-tests/memo/memoize_tests.ml

```diff
@@ -704,12 +704,12 @@ let%expect_test "diamond with non-uniform cutoff structure" =
   [%expect
     {|
     Evaluating count_runs: 2
-    Starting evaluating yes_cutoff
-    Evaluated yes_cutoff: 1
     Starting evaluating no_cutoff
     Evaluated no_cutoff: 1
     Starting evaluating after_no_cutoff
     Evaluated after_no_cutoff: 2
+    Starting evaluating yes_cutoff
+    Evaluated yes_cutoff: 1
```

Comment:
This change is a bit of a mystery. Apparently we used to record the pair of dependencies `yes_cutoff` and `no_cutoff` in the opposite order.